packages/block-library/src/terms-query/edit/inspector-controls/index.js

```diff
@@ -46,24 +46,26 @@ export default function TermsQueryInspectorControls( {
 		( _taxonomy ) => _taxonomy.slug === taxonomy
 	)?.hierarchical;
 
-	const isTaxonomyMatchingTemplate = templateSlug?.startsWith(
-		// `Tags` are a special case in WP template hierarchy.
-		taxonomy === 'post_tag' ? 'tag' : taxonomy
-	);
-
-	// Only display the inherit control if the taxonomy is hierarchical and matches the current template.
+	// Display the inherit control when we're in a taxonomy-related
+	// template (category, tag, or custom taxonomy).
 	const displayInheritControl =
-		isTaxonomyHierarchical && isTaxonomyMatchingTemplate;
+		[ 'taxonomy', 'category', 'tag', 'archive' ].includes( templateSlug ) ||
```

Comment:
Nice. Why do we want it for archive templates? I didn't see it in the `is_tax() || is_category() || is_tag()` check above.

Reply:
No, there's no issue. Just not sure I fully follow. You mean there could be archive template that have a taxonomy as the query object? 🤔